csharp/src/Ice/SslTransceiver.cs

```diff
@@ -446,203 +445,117 @@ private void ReadCompleted(IAsyncResult result)
-                                message += "\ncertificate revoked (ignored)";
-                            }
-                        }
-                        else if (status.Status == X509ChainStatusFlags.OfflineRevocation)
-                        {
-                            // If a certificate's revocation status cannot be determined, the strictest policy is to
-                            // reject the connection.
-                            if (_engine.CheckCRL > 1)
-                            {
-                                message += "\ncertificate revocation list is offline";
-                                ++errorCount;
-                            }
-                            else
-                            {
-                                message += "\ncertificate revocation list is offline (ignored)";
-                            }
-                        }
-                        else if (status.Status == X509ChainStatusFlags.RevocationStatusUnknown)
-                        {
-                            // If a certificate's revocation status cannot be determined, the strictest policy is to
-                            // reject the connection.
-                            if (_engine.CheckCRL > 1)
-                            {
-                                message += "\ncertificate revocation status unknown";
-                                ++errorCount;
-                            }
-                            else
-                            {
-                                message += "\ncertificate revocation status unknown (ignored)";
-                            }
-                        }
-                        else if (status.Status == X509ChainStatusFlags.PartialChain)
-                        {
-                            message += "\npartial certificate chain";
-                            ++errorCount;
-                        }
-                        else if (status.Status != X509ChainStatusFlags.NoError)
-                        {
-                            message = message + "\ncertificate chain error: " + status.Status.ToString();
-                            ++errorCount;
-                        }
+                        chainStatus.Remove(
+                            chainStatus.Find(status => status.Status == X509ChainStatusFlags.UntrustedRoot));
+                        errors ^= SslPolicyErrors.RemoteCertificateChainErrors;
                     }
-
-                    if (errorCount == 0)
+                    else if (!chainStatus.Exists(status => status.Status == X509ChainStatusFlags.UntrustedRoot))
                     {
-                        errors ^= (int)SslPolicyErrors.RemoteCertificateChainErrors;
+                        chainStatus.Add(new X509ChainStatus() { Status = X509ChainStatusFlags.UntrustedRoot });
+                        errors |= SslPolicyErrors.RemoteCertificateChainErrors;
                     }
                 }
```

Comment:
To me the content of this `if(caCerts != null)` block should go after the Build call on the custom chain.

Reply:
I think this is already the case